src/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -104,10 +105,135 @@ internal T WithExtraQueryParameters(string extraQueryParameters)
         /// to applications managing several accounts (like a mail client with several mailboxes)
         /// </summary>
         /// <param name="authorityUri">Uri for the authority</param>
+        /// <param name="validateAuthority"></param>
         /// <returns>The builder to chain the .With methods</returns>
-        public T WithAuthorityOverride(string authorityUri)
+        public T WithAuthority(Uri authorityUri, bool validateAuthority = false)
```

Comment:
>false [](start = 74, length = 5)

I'm wondering if the default should be true? Except for B2C?

Reply:
yeah, I'm not sure either.  This was false by default before, and this could theoretically be a B2C authority as well as AAD, or ADFS.  So I think the best back-compat thing to do is have this one be false and hope people move to the more explicit APIs going forwards.

---
In reply to: [250002974](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/815#discussion_r250002974) [](ancestors = 250002974)

Reply:
Only "worry" I have is that it's not as secure by default as possible. I would consider making the default true